client/src/main/java/org/eclipse/hono/client/impl/CredentialsClientImpl.java

```diff
@@ -39,7 +39,7 @@
  * A Vertx-Proton based client for Hono's Credentials API.
  *
  */
-public final class CredentialsClientImpl extends AbstractRequestResponseClient<CredentialsResult<CredentialsObject>> implements CredentialsClient {
+public class CredentialsClientImpl extends AbstractRequestResponseClient<CredentialsResult<CredentialsObject>> implements CredentialsClient {
```

Comment:
My understanding is that you want to change this in order to be able to extend the class and add your custom operations. If that is the case, then we should make the existing methods of `CredentialsClientImpl` final in order to prevent subclasses from overriding them.

Reply:
I agree. Will do.

Reply:
Done